test/extended/image_ecosystem/plugin.go

```diff
@@ -286,16 +286,30 @@ func (j *JenkinsRef) startJob(jobName string) *JobMon {
 
 // Returns the content of a Jenkins job XML file. Instances of the
 // string "PROJECT_NAME" are replaced with the specified namespace.
-func (j *JenkinsRef) readJenkinsJob(filename, namespace string) string {
+// Variables named in the vars map will also be replaced with their
+// corresponding value.
+func (j *JenkinsRef) readJenkinsJobUsingVars(filename, namespace string, vars map[string]string) string {
 	pre := exutil.FixturePath("testdata", "jenkins-plugin", filename)
 	post := exutil.ArtifactPath(filename)
-	err := exutil.VarSubOnFile(pre, post, "PROJECT_NAME", namespace)
+
+	if vars == nil {
+		vars = map[string]string{}
+	}
+	vars["PROJECT_NAME"] = namespace
```

Comment:
do we want to use a more well-delimited string like "${PROJECT_NAME}" to avoid  inadvertent substitution?